aws_lambda_powertools/utilities/data_masking/base.py

```diff
@@ -26,7 +26,72 @@
 
 logger = logging.getLogger(__name__)
 
-
+def prepare_data(data: Any, _visited: set[int] | None = None) -> Any:
```

Comment:
Ruff is complaining that this function is complex with too many returns (https://docs.astral.sh/ruff/rules/too-many-return-statements/). Although I understand that returning early avoids `if` checks and stuff like that. Do you see room to improve this function? If you want, I can try to optimize this code.

Reply:
Thanks for resolving the comments @VatsalGoel3! Let me work in this function to see if I can improve it.